indigo_api/models/tasks.py

```diff
@@ -183,9 +184,35 @@ def decorate_potential_assignees(cls, tasks, country):
             elif task.state == 'pending_review':
                 task.potential_assignees = [u for u in potential_reviewers if task.assigned_to_id != u.id and
                                             (u.has_perm('indigo_api.close_any_task') or task.submitted_by_user_id != u.id)]
+            else:
+                task.potential_assignees = []
+
+            # move the current user first
+            if task.potential_assignees and current_user and current_user.is_authenticated:
+                task.potential_assignees.sort(key=lambda u: 0 if u.id == current_user.id else 1)
+
+        # mark users that have too many tasks
+        cls.decorate_user_task_limits(set(u for task in tasks for u in task.potential_assignees), tasks)
 
         return tasks
 
+    @classmethod
+    def decorate_user_task_limits(cls, users, tasks):
+        """ Count assigned tasks for these users and decorate users that have exceeded their threshold.
+        """
+        states = set(t.state for t in tasks)
```

Comment:
I'm not sure about this. Wouldn't it change `counts` to allow e.g. a reviewer to pick up more tasks if they filter the tasks listing view to only include `pending_review` tasks? or is this decoration all happening before the tasks are filtered? (seems odd if true) 
I guess regardless, I'm a little puzzled by what `states` have to do with `counts`?

Reply:
The two assignable states we care about are "open" and "pending review". We consider them to have separate task counters. So if the limit of open tasks is 5, they can be assigned 5 open and 5 pending review. That's part of why we take states into account. (I'm open to changing this, I don't have a particularly strong argument for keeping it this way).

In most cases, this function only actually gets a single task and it's simple. The tricky situation is when a user is trying to re-assign multiple tasks through the task listing view, and they choose tasks across both "open" and "pending review". To avoid dodging the limits, we need to effectively count all assigned tasks together.